controllers/idler/idler_controller_test.go

```diff
@@ -256,6 +266,95 @@ func TestEnsureIdling(t *testing.T) {
+		reconciler, req, cl, _ := prepareReconcile(t, idler.Name, newGetHostClusterReady, idler, nsTmplSet, mur)
+		idlerTimeoutPlusOneSecondAgo := time.Now().Add(-time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
+		preparePayloads(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo)
+
+		// first reconcile to track pods
+		res, err := reconciler.Reconcile(context.TODO(), req)
+		require.NoError(t, err)
+		assert.True(t, res.Requeue)
+		assert.Equal(t, int(res.RequeueAfter), 0)
+
+		// second reconcile should delete pods and create notification
+		res, err = reconciler.Reconcile(context.TODO(), req)
+		//then
+		assert.NoError(t, err)
+		memberoperatortest.AssertThatIdler(t, idler.Name, cl).
+			HasConditions(memberoperatortest.Running(), memberoperatortest.IdlerNotificationCreated())
+		//check the notification is actually created
+		hostCl, _ := reconciler.GetHostCluster()
+		notification := &toolchainv1alpha1.Notification{}
+		err = hostCl.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: test.HostOperatorNs,
+			Name:      "alex-stage-idled",
+		}, notification)
+		require.NoError(t, err)
+
+	})
+
+	t.Run("Pods idled but notification not created", func(t *testing.T) {
+		idler := &toolchainv1alpha1.Idler{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "alex-dev",
+				Labels: map[string]string{
+					toolchainv1alpha1.OwnerLabelKey: "alex",
+				},
+			},
+			Spec: toolchainv1alpha1.IdlerSpec{TimeoutSeconds: 60},
+		}
+		namespaces := []string{"dev", "stage"}
+		usernames := []string{"alex"}
+		nsTmplSet := newNSTmplSet(test.MemberOperatorNs, "alex", "advanced", "abcde11", namespaces, usernames)
+		reconciler, req, cl, _ := prepareReconcile(t, idler.Name, newGetHostClusterReady, idler, nsTmplSet) // mur not added
+		idlerTimeoutPlusOneSecondAgo := time.Now().Add(-time.Duration(idler.Spec.TimeoutSeconds+1) * time.Second)
+		preparePayloads(t, reconciler, idler.Name, "todelete-", idlerTimeoutPlusOneSecondAgo)
+
+		// first reconcile to track pods
+		res, err := reconciler.Reconcile(context.TODO(), req)
+		require.NoError(t, err)
+		assert.True(t, res.Requeue)
+		assert.Equal(t, int(res.RequeueAfter), 0)
+		// second reconcile should delete pods and create notification
+		res, err = reconciler.Reconcile(context.TODO(), req)
+		//then
+		assert.NoError(t, err)
+		memberoperatortest.AssertThatIdler(t, idler.Name, cl).
+			HasConditions(memberoperatortest.Running(), memberoperatortest.IdlerNotificationCreationFailed("Could not get the MUR with name: alex "))
+		hostCl, _ := reconciler.GetHostCluster()
+		notification := &toolchainv1alpha1.Notification{}
+		err = hostCl.Client.Get(context.TODO(), types.NamespacedName{
+			Namespace: test.HostOperatorNs,
+			Name:      "alex-dev-idled",
+		}, notification)
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "\"alex-dev-idled\" not found")
+
+		t.Run("third reconcile doesn't create notification as no pods to idle", func(t *testing.T) {
```

Comment:
Do we really need another reconcile? Is this step just supposed to verify the idler status is set correctly? If so wouldn't the status be set after the 2nd reconcile so we could just add that check right after the 2nd reconcile?

Reply:
This entire test case has been removed, it did not really add anything to the coverage. done in a58c63e